students/JeffBennett/Lesson05/mailroom_3.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+import datetime
+import sys
+
+# Data
+
+donors = {
+    'Lorenzo Currie': [3000, 250, 5000],
+    'Marie Kane': [1000, 195, 50.50],
+    'John Beresford Tipton': [10000, 3000, 2500],
+    'Janna Adams': [35.50, 19.75],
+    'LeSean Haskell': [100],
+    'Laila Samir': [1000, 250]
+}
+
+letter_to_all_donors = '''\nDear {}\n\n
+I write to thank you again for all the support you have provided to Northwest
+Lifeboats.  During the past year your total gifts of ${:,.2f} have greatly
+furthered our activities.
+\n\nI have exciting news that one of our prospective donors may be prepared
+to match your next donation, which would double your impact on our efforts.
+I will be back in touch with you soon with more details.\n\nSincerely yours
+\n\nJ A Bennett\nDirector'''
+
+# Input/Output
+
+
+def print_main_menu():
+    """print main menu and prompt for user choice."""
+    print('\nMain Menu\n' + '-'*25 +
+          '\n(1)\tSend a Thank You\n(2)\tCreate a Report'
+          '\n(3)\tSend Letters to All Donors\n(4)\tQuit')
+
+
+def print_donor_list(donors):
+    """print list of donor names."""
+    print('\nOur Current List of Donors:\n' + '-' * 25)
+    for key in donors:
+        print(key)
+
+
+def input_gift():
+    """input donor gift."""
+    while True:
+        try:
+            gift = float(input("\nEnter amount of gift: " +
+                               "or '0' to return to the main menu: "))
+            return gift
```

Comment:
You generally only want to include the code in the try block that might raise the error, so I'd remove the return statement and place it after the except block at the end of the function